File: LeetCode/Java/1108. Defanging an IP Address.java
class Solution {
    public String defangIPaddr(String address) {
        StringBuilder defangedAddress = new StringBuilder();
        
        for (int i = 0; i < address.length(); i++) {
            char currentChar = address.charAt(i);
            
            if (currentChar == '.')
                defangedAddress.append("[.]");
            else
                defangedAddress.append(currentChar);
        }
        
        return defangedAddress.toString();
    }
}

File: LeetCode/Python/1108. Defanging an IP Address.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        defange_address = []
        
        for char in address:
            if char == '.':
                defange_address.append("[.]")
            else:
                defange_address.append(char)
                
        return "".join(defange_address)
